### DISCUSSION TITLE
conda environment

### DISCUSSION THREAD

**Discussion by FHQ0113** [idaholab/moose#28914] [Q&A General]
### Bug Description

```pre
(moose) haoqi-moose@pwmoose-PowerEdge-T640:~/projects$ git clone https://github.com/idaholab/moose.git
Cloning into 'moose'...
fatal: unable to access 'https://github.com/idaholab/moose.git/': GnuTLS recv error (-54): Error in the pull function.
(moose) haoqi-moose@pwmoose-PowerEdge-T640:~/projects$ git clone https://github.com/idaholab/moose.git
Cloning into 'moose'...
fatal: unable to access 'https://github.com/idaholab/moose.git/': GnuTLS recv error (-110): The TLS connection was non-properly terminated.
(moose) haoqi-moose@pwmoose-PowerEdge-T640:~/projects$ git clone https://github.com/idaholab/moose.git
Cloning into 'moose'...
fatal: unable to access 'https://github.com/idaholab/moose.git/': Failed to connect to github.com port 443: Connection timed out
```

### Steps to Reproduce

(https://mooseframework.inl.gov/getting_started/installation/conda.html)
I encountered the following problem when using the above connection to create and install MOOSE on a Linux system:

### Impact

Causing me a lot of trouble in completing my work

**Comment by milljm** [comment]
This was not something that could be solved with a change to MOOSE's code (an issue). Converting to a discussion.


It looks as though the company, or organization you are with, blocks your access to https://github.com (and probably all outbound sites via the terminal), by means of first requiring you to use their ROOT Certificates. Just like us, here at INL. This is basically so they can watch what you're doing (which is called an MITM attack, if it were an actual threat actor). Stating this, just in case you are actually falling victim to an MITM... Please look into this, if you are in fact using a personal machine.

I see this error when one does not have their certificates properly set up.

Unfortunately, there is nothing anyone here can do to help you other than your own organization's network team. They should be providing you with the certificates, and how to enable them within your terminal. If you know what you are doing you should be able to extrapolate what your favorite web browser is doing when navigating to https://github.com, and extract that certificate to somewhere on your drive, and export a few influential environment variables to let `git` know where they are. My exact setup to show you what those environment variables are:
```bash
❯ env | grep -i root
REQUESTS_CA_BUNDLE=/opt/certs/CAINLROOT.crt
SSL_CERT_FILE=/opt/certs/CAINLROOT.crt
CURL_CA_BUNDLE=/opt/certs/CAINLROOT.crt
```

Your network team should be providing you with the `*.crt` file. Anyways... One work-around that _might_ work, and should not be considered a "fix" or used for extended amounts of time, is to bypass SSL verification temporarily:
```bash
GIT_SSL_NO_VERIFY=true git clone https://github.com/idaholab/moose
```
This effectually turns off SSL protections. The errors you are seeing are actually "good" errors. The program is trying to alert to you something is wrong with your _secure connection_. That there isn't one!

The other thing you can do is to not use HTTPs protocol, and instead create an SSH public/private key pair, and clone MOOSE that way: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Using SSH is the preferred way to clone MOOSE anyway, if you plan on making changes and commiting them back MOOSE.
